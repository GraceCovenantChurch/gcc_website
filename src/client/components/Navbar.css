
#mainNavbar {
  border: 0;
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;
  margin: 1em 3%;

  .navbar-brand {
    img {
      height: 85px;
    }

    .logo-light {
      display: block;
    }

    .logo-dark {
      display: none;
    }

    @media(max-width: 991px) {
      .logo-light {
        display: none;
      }
      .logo-dark {
        display: block;
      }
    }
  }

  .navbar-nav {
    .nav-link {
      text-align: center;
      font-size: 18px;
      margin-left: 15px;
      margin-right: 15px;
      font-weight: 500;
    }
  }

  .navbar-toggler {
    border: none;
    &:focus {
      outline: none;
    }
  }

  &.navbar-light {
    .navbar-toggler-icon {
      background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='black' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E");
      &:hover {
        background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='gray' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E");
      }
    }

    .navbar-nav {
      .nav-link, a {
        color: white;
        &:hover {
          color: #cccccc;
          text-decoration: none;
        }
      }
    }
  }

  @media(max-width: 991px) {
    position: fixed;
    top: 0px !important;
    .aboutLink {
      display: inline;
    }
    &.navbar-light {
      background-color: white;
      box-shadow: 0px 0px 2px 2px rgba(0,0,0,0.2);

      .navbar-nav {
        .nav-link, a {
          display: flex;
          justify-content: center;
          color: black;
          &:hover {
            color: #cccccc;
            text-decoration: none;
          }
        }
      }
    }
  }

  transition: top ease 0.3s;
}

.aboutLink {
  display: none;
}