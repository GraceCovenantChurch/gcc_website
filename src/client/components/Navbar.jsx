import React, {Component} from 'react';
import Link from 'react-router-dom/Link';
import classnames from 'classnames';
import { PAGE_LOADED } from '../modules/page';
const styles = (typeof CSS !== 'undefined') && require('./Navbar.css');

class Navbar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      collapsed: true,
      collapsing: false,
      in: false,
      height: 'auto',
      atTop: true,
      hidden: true,
      hiding: false,
      top: -100,
    };
  }

  componentWillMount() {
    this.scollCallback = this.handleScroll.bind(this);
    window.addEventListener('scroll', this.scollCallback);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.scollCallback);
  }

  componentDidMount() {
    this.handleScroll();
  }

  toggle() {
    if (this.state.collapsing) {
      return;
    }

    let collapsed = this.state.collapsed;
    this.setState({
      collapsed: !collapsed,
      collapsing: true,
      height: collapsed ? 0 : this.refs['navbar-nav'].clientHeight,
    }, () => {
      setTimeout(() => {
        this.setState({
          height: collapsed ? this.refs['navbar-nav'].clientHeight : 0,
        }, () => {
          setTimeout(() => {
            this.setState({
              collapsing: false,
              height: 'auto',
              in: collapsed ? true : undefined,
            })
          }, 300);
        });
      }, 10);
    });
  }

  open() {
    if (this.state.collapsed) {
      this.toggle();
    }
  }

  close() {
    if (!this.state.collapsed) {
      this.toggle();
    }
  }

  handleScroll(e) {
    this.setState({
      atTop: window.scrollY < 200,
    });

    let hidden = this.state.hidden;
    let shouldHide = window.scrollY < 800;
    if (hidden !== shouldHide && !this.state.hiding) {
      if (!shouldHide) {
        this.setState({
          hiding: true,
          hidden: false,
          top: -100,
        }, () => {
          setTimeout(() => {
            this.setState({
              top: 0
            }, () => {
              this.setState({
                hiding: false,
              });
            });
          }, 100)
        })
      } else {
        this.setState({
          hiding: true,
          top: 0,
        }, () => {
          setTimeout(() => {
            this.setState({
              top: -100
            }, () => {
              setTimeout(() => {
                this.setState({
                  hiding: false,
                  hidden: true,
                });
              }, 300);
            });
          }, 100);
        });
      }
    }
  }

  render() {
    return (
      <nav id="navbar" className={classnames(`navbar navbar-expand-lg ${(this.props.className || '')}`, {
        popup: !this.state.atTop && !this.state.hidden,
      })} style={{
        top: this.state.atTop ? '0px' : this.state.top,
      }}>
        {React.cloneElement(this.props.brand, {
          className: "navbar-brand",
          onClick: this.close.bind(this),
        })}
        <button className="navbar-toggler" type="button" onClick={this.toggle.bind(this)}>
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className={classnames('navbar-collapse', {
            show: this.state.in,
            collapsing: this.state.collapsing,
            collapse: !this.state.collapsing,
          })} style={{
            height: this.state.height,
          }}>
          <ul className="navbar-nav ml-auto" ref="navbar-nav">
            {React.Children.map(this.props.links, link => {
              return (
                <li className="nav-item">
                  {React.cloneElement(link, {
                    onClick: this.close.bind(this),
                    className: 'nav-link',
                  })}
                </li>
              );
            })}
          </ul>
        </div>
      </nav>
    );
  }
}

export default Navbar;
