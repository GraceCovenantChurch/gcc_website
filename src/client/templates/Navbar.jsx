import React from 'react';
import Link from 'react-router-dom/Link';

const Navbar = () => {
  return (
    <div>
      <Link to="/">Home</Link><br />
      <Link to="/page">Page</Link><br />
      <Link to="/pages/async">Async Page</Link><br />
    </div>
  );
}

export default Navbar;
