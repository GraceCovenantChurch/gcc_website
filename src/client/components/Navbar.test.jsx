/* global mount, expect */

import React from 'react';
import toJSON from 'enzyme-to-json';
import Navbar from './Navbar';

describe('Navbar component', () => {
  it('opens and closes', (done) => {
    const navbar = mount(<Navbar
      brand={<a href="/">Brand</a>}
      links={[
        <a href="/">a</a>,
        <a href="/">b</a>,
        <a href="/">c</a>,
      ]}
    />);
    expect(toJSON(navbar)).toMatchSnapshot();

    const toggler = navbar.find('.navbar-toggler').at(0);

    // open nav
    toggler.simulate('click');
    navbar.update();
    expect(toJSON(navbar)).toMatchSnapshot();

    setTimeout(() => {
      // wait to open
      navbar.update();
      expect(toJSON(navbar)).toMatchSnapshot();

      // close nav
      toggler.simulate('click');
      navbar.update();
      expect(toJSON(navbar)).toMatchSnapshot();

      setTimeout(() => {
        // wait to close
        navbar.update();
        expect(toJSON(navbar)).toMatchSnapshot();
        done();
      }, 400);
    }, 400);
  });
});
