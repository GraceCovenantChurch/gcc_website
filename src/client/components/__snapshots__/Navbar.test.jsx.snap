// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component opens and closes 1`] = `
<Navbar
  brand={
    <a
      href="/"
    >
      Brand
    </a>
  }
  className=""
  links={
    Array [
      <a
        href="/"
      >
        a
      </a>,
      <a
        href="/"
      >
        b
      </a>,
      <a
        href="/"
      >
        c
      </a>,
    ]
  }
>
  <nav
    className="navbar navbar-expand-lg "
    id="Navbar__mainNavbar"
  >
    <a
      className="navbar-brand Navbar__navbar-brand"
      href="/"
    >
      Brand
    </a>
    <button
      className="navbar-toggler Navbar__navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon Navbar__navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <ul
        className="navbar-nav ml-auto Navbar__navbar-nav"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            a
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            b
          </a>
        </li>
        <li
          className="nav-item"
          key=".2"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            c
          </a>
        </li>
      </ul>
    </div>
  </nav>
</Navbar>
`;

exports[`Navbar component opens and closes 2`] = `
<Navbar
  brand={
    <a
      href="/"
    >
      Brand
    </a>
  }
  className=""
  links={
    Array [
      <a
        href="/"
      >
        a
      </a>,
      <a
        href="/"
      >
        b
      </a>,
      <a
        href="/"
      >
        c
      </a>,
    ]
  }
>
  <nav
    className="navbar navbar-expand-lg "
    id="Navbar__mainNavbar"
  >
    <a
      className="navbar-brand Navbar__navbar-brand"
      href="/"
    >
      Brand
    </a>
    <button
      className="navbar-toggler Navbar__navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon Navbar__navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapsing"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <ul
        className="navbar-nav ml-auto Navbar__navbar-nav"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            a
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            b
          </a>
        </li>
        <li
          className="nav-item"
          key=".2"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            c
          </a>
        </li>
      </ul>
    </div>
  </nav>
</Navbar>
`;

exports[`Navbar component opens and closes 3`] = `
<Navbar
  brand={
    <a
      href="/"
    >
      Brand
    </a>
  }
  className=""
  links={
    Array [
      <a
        href="/"
      >
        a
      </a>,
      <a
        href="/"
      >
        b
      </a>,
      <a
        href="/"
      >
        c
      </a>,
    ]
  }
>
  <nav
    className="navbar navbar-expand-lg "
    id="Navbar__mainNavbar"
  >
    <a
      className="navbar-brand Navbar__navbar-brand"
      href="/"
    >
      Brand
    </a>
    <button
      className="navbar-toggler Navbar__navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon Navbar__navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse show collapse"
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <ul
        className="navbar-nav ml-auto Navbar__navbar-nav"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            a
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            b
          </a>
        </li>
        <li
          className="nav-item"
          key=".2"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            c
          </a>
        </li>
      </ul>
    </div>
  </nav>
</Navbar>
`;

exports[`Navbar component opens and closes 4`] = `
<Navbar
  brand={
    <a
      href="/"
    >
      Brand
    </a>
  }
  className=""
  links={
    Array [
      <a
        href="/"
      >
        a
      </a>,
      <a
        href="/"
      >
        b
      </a>,
      <a
        href="/"
      >
        c
      </a>,
    ]
  }
>
  <nav
    className="navbar navbar-expand-lg "
    id="Navbar__mainNavbar"
  >
    <a
      className="navbar-brand Navbar__navbar-brand"
      href="/"
    >
      Brand
    </a>
    <button
      className="navbar-toggler Navbar__navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon Navbar__navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse show collapsing"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <ul
        className="navbar-nav ml-auto Navbar__navbar-nav"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            a
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            b
          </a>
        </li>
        <li
          className="nav-item"
          key=".2"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            c
          </a>
        </li>
      </ul>
    </div>
  </nav>
</Navbar>
`;

exports[`Navbar component opens and closes 5`] = `
<Navbar
  brand={
    <a
      href="/"
    >
      Brand
    </a>
  }
  className=""
  links={
    Array [
      <a
        href="/"
      >
        a
      </a>,
      <a
        href="/"
      >
        b
      </a>,
      <a
        href="/"
      >
        c
      </a>,
    ]
  }
>
  <nav
    className="navbar navbar-expand-lg "
    id="Navbar__mainNavbar"
  >
    <a
      className="navbar-brand Navbar__navbar-brand"
      href="/"
    >
      Brand
    </a>
    <button
      className="navbar-toggler Navbar__navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon Navbar__navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <ul
        className="navbar-nav ml-auto Navbar__navbar-nav"
      >
        <li
          className="nav-item"
          key=".0"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            a
          </a>
        </li>
        <li
          className="nav-item"
          key=".1"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            b
          </a>
        </li>
        <li
          className="nav-item"
          key=".2"
        >
          <a
            className="nav-link Navbar__nav-link undefined"
            href="/"
            onClick={[Function]}
          >
            c
          </a>
        </li>
      </ul>
    </div>
  </nav>
</Navbar>
`;
